### PR TITLE
[Model] Modify 'tutorial group' model to support letter

### LINKED ISSUE #63: [Model] Modify 'tutorial group' model to support letter

### PR DESCRIPTION
**Pull request by kevinmingtarja**
Closes #63

### CODE REVIEW

File: src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java [hunk 1]
```diff
@@ -45,30 +45,26 @@ public void equals() {
...
+        assertTrue(predicate.test(new PersonBuilder().withTutorialGroup("T19").build()));
 
         // Only one matching keyword
-        predicate = new TutorialGroupContainsKeywordsPredicate(Arrays.asList("9123456", "908976"));
-        assertTrue(predicate.test(new PersonBuilder().withTutorialGroup("908976").build()));
-
-        // Non-matching keyword
-        predicate = new TutorialGroupContainsKeywordsPredicate(Arrays.asList("908976", "9123456"));
-        assertTrue(predicate.test(new PersonBuilder().withTutorialGroup("9123456908976").build()));
+        predicate = new TutorialGroupContainsKeywordsPredicate(Arrays.asList("T19", "908976"));
```
Comment (choonyongchan) [inline src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java:52]: Perhaps the "908976" can be changed to follow the "[A-Z]\\d{2}" format.

File: src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java [hunk 2]
```diff
@@ -45,30 +45,26 @@ public void equals() {
...
         // Zero keywords
         TutorialGroupContainsKeywordsPredicate predicate =
                 new TutorialGroupContainsKeywordsPredicate(Collections.emptyList());
-        assertFalse(predicate.test(new PersonBuilder().withTutorialGroup("908976").build()));
+        assertFalse(predicate.test(new PersonBuilder().withTutorialGroup("T19").build()));
 
-        // Keywords does not match name
+        // Keywords does not match tutorial group
         predicate = new TutorialGroupContainsKeywordsPredicate(
                 Arrays.asList("99999", "123452354", "12345235", "12345253"));
```
Comment (choonyongchan) [inline src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java:65]: Perhaps we can follow the "[A-Z]\d{2}" format here.
predicate = new TutorialGroupContainsKeywordsPredicate(
                Arrays.asList("T20", "T21", "T22", "T23"));

File: src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java [hunk 3]
```diff
@@ -45,30 +45,26 @@ public void equals() {
...
+                new TutorialGroupContainsKeywordsPredicate(Collections.singletonList("T19"));
+        assertTrue(predicate.test(new PersonBuilder().withTutorialGroup("T19").build()));
 
         // Only one matching keyword
-        predicate = new TutorialGroupContainsKeywordsPredicate(Arrays.asList("9123456", "908976"));
-        assertTrue(predicate.test(new PersonBuilder().withTutorialGroup("908976").build()));
-
-        // Non-matching keyword
-        predicate = new TutorialGroupContainsKeywordsPredicate(Arrays.asList("908976", "9123456"));
-        assertTrue(predicate.test(new PersonBuilder().withTutorialGroup("9123456908976").build()));
```
Comment (choonyongchan) [inline src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java:57]: Perhaps you may consider if this is useful?
predicate = new TutorialGroupContainsKeywordsPredicate(Arrays.asList("T12", "T13"));
assertFalse(predicate.test(new PersonBuilder().withTutorialGroup("T12T13").build()));

Comment (kevinmingtarja) [reply src/test/java/seedu/address/model/person/TutorialGroupContainsKeywordsPredicateTest.java:57]: Hm I think `T12T13` will cause an exception to be thrown due to the TG format

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 4]
```diff
@@ -26,22 +26,22 @@
...
-                    new Nationality("Indonesia"), new TutorialGroup("10"), new SocialHandle("@berny"),
+                    new Nationality("Indonesia"), new TutorialGroup("T10"), new SocialHandle("@berny"),
                     new Gender("F"), new Remark("likes to dance"), getTagSet("colleagues", "friends")),
             new Person(new Name("Charlotte Oliveiro"), new Phone("93210283"), new Email("charlotte@example.com"),
-                    new Nationality("Malaysia"), new TutorialGroup("28"), new SocialHandle("@charlotte"),
+                    new Nationality("Malaysia"), new TutorialGroup("W28"), new SocialHandle("@charlotte"),
                     new Gender("F"), new Remark("likes to eat"), getTagSet("neighbours")),
             new Person(new Name("David Li"), new Phone("91031282"), new Email("lidavid@example.com"),
-                    new Nationality("Singapore"), new TutorialGroup("02"), new SocialHandle("@davidl"),
+                    new Nationality("Singapore"), new TutorialGroup("E02"), new SocialHandle("@davidl"),
```
Comment (hsiaotingluv) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:38]: E technically isn't a valid tutorial group name right?

File: src/test/java/seedu/address/model/person/TutorialGroupTest.java [hunk 5]
```diff
@@ -25,11 +25,13 @@ public void isValidTutorialGroup() {
         assertThrows(NullPointerException.class, () -> TutorialGroup.isValidTutorialGroup(null));
 
         // invalid tutorial group
-        assertFalse(TutorialGroup.isValidTutorialGroup("B")); // contains character
+        assertFalse(TutorialGroup.isValidTutorialGroup("B")); // one character only
+        assertFalse(TutorialGroup.isValidTutorialGroup("09")); // two digits only
+        assertFalse(TutorialGroup.isValidTutorialGroup("T124")); // length exceeds maximum
+        assertFalse(TutorialGroup.isValidTutorialGroup("t09")); // lowercase character
```
Comment (hsiaotingluv) [inline src/test/java/seedu/address/model/person/TutorialGroupTest.java:31]: was thinking maybe we could make it case insensitive hahah, more convenient right?

File: src/test/java/seedu/address/model/person/TutorialGroupTest.java [hunk 6]
```diff
@@ -25,11 +25,20 @@ public void isValidTutorialGroup() {
...
-        assertTrue(TutorialGroup.isValidTutorialGroup("9")); // one digit
-        assertTrue(TutorialGroup.isValidTutorialGroup("")); // empty string
+        assertTrue(TutorialGroup.isValidTutorialGroup("M09")); // M/T/W/F and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("T09")); // M/T/W/F and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("W20")); // M/T/W/F and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("F20")); // M/T/W/F and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("m12")); // m/t/w/f and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("t09")); // m/t/w/f and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("w09")); // m/t/w/f and two digits
+        assertTrue(TutorialGroup.isValidTutorialGroup("f10")); // m/t/w/f and two digits
```
Comment (hsiaotingluv) [inline src/test/java/seedu/address/model/person/TutorialGroupTest.java:42]: nice one!!

**Pull request merged**